public/js/controllers/charts_controller.js

```diff
@@ -320,11 +354,21 @@ export default class extends Controller {
           }
         }
         break
+      case 'stake-participation':
+        d = percentStakedFunc(data, isHeightAxis, isDayBinned)
+        assign(gOptions, mapDygraphOptions(d[0], [xlabel, 'Stake Participation', 'Ticket Pool Value',
+          'Coin Supply'], true, 'Stake Participation (%)', undefined, true, false))
+        gOptions.valueRange = [null, d[1] + 5]
```

Comment:
This is why I mentioned the bug here: https://github.com/decred/dcrdata/pull/1359#discussion_r288960156 The fix in #1359 will work effectively on this chart too thus correcting that chart you were able to create by messing with the rangeselector.

I thought it wise not to repeat the same code in this PR so as not to have many merge conflicts while merging both PRs.

Reply:
>The hiccup here is that the data supplied to the legendFormatter doesn't include the data point's index in the series, so d[2] would need to be associative, i.e. d[2] = { 'x1': y1, 'x2': y2, ... } which adds a little time to the lookup.

I don't think we should take this route since the value range fix works.

Reply:
You are still tripling the rendering time. Loading blocks-based data on this PR's chart takes about 40 seconds for me. Compare that to ~ 7 seconds for other blocks-based charts where there isn't extraneous tripling of data. 

`valueRange` should simply be avoided, as it's implementation is incomplete.

Reply:
I still don't know how 40 sec or 7 sec were obtained but on my end I get much lower time with the same data.

**Here is a screenshot of my terminal.**
![Screenshot from 2019-06-04 12-45-44](https://user-images.githubusercontent.com/22055953/58869398-d06d9580-86c6-11e9-871d-5aef38988fd3.png)

**Screenshot of the browser metrics.**
![Screenshot from 2019-06-04 12-34-57](https://user-images.githubusercontent.com/22055953/58869434-dfecde80-86c6-11e9-9a44-43f1b1907985.png)

Reply:
>Possible solution: Pass the poolValue, coinSupply values back from percentStakedFunc as a separate structure. Set it as a custom option on `gOptions.myCustomOptionName = d[2]`, and then use `Dygraph.getOption` to retrieve the information during legend formatting.

I am also curious to know how you got `Dygraph.getOption` working because its returning errors.
![Screenshot from 2019-06-04 13-05-35](https://user-images.githubusercontent.com/22055953/58870777-7f12d580-86c9-11e9-92cf-7d1473cb3269.png)

Also from the API, legendformatter doesn't pass the current graph instance: https://github.com/danvk/dygraphs/pull/683

Reply:
Yeah you may need to make it a controller method. 

The timings are for rendering, not downloading. For blocks data, the download is a pretty small fraction of the time to display. Most of the time is locked in Dygraphs data processing and drawing. That's why adding a lot of `file` data that you aren't actually displaying is a bad idea. Dygraphs is trying to draw ~700,000 extra line segments that you are just hiding with `valueRange`.